examples/pytorch_lightning_simple.py

```diff
@@ -0,0 +1,190 @@
+
+    def __init__(self, model, optimizer_name, lr):
+        super(LightningNet, self).__init__()
+
+        # Be careful not to overwrite `pl.LightningModule` attributes such as `self.model`.
+        self._model = model
+        self._optimizer_name = optimizer_name
+        self._lr = lr
+
+    def forward(self, data):
+        return self._model(data)
+
+    def training_step(self, batch, batch_nb):
+        data, target = batch
+        output = self.forward(data)
+        loss = F.nll_loss(output, target)
+        return {'loss': loss}
+
+    def validation_step(self, batch, batch_nb):
+        data, target = batch
+        output = self.forward(data)
+        pred = output.argmax(dim=1, keepdim=True)
+        correct = pred.eq(target.view_as(pred)).sum().item()
+        accuracy = correct / data.size(0)
+        return {'validation_accuracy': accuracy}
+
+    def validation_end(self, outputs):
+        accuracy = sum(x['validation_accuracy'] for x in outputs) / len(outputs)
+        # Pass the accuracy to the `DictLogger` via the `'log'` key.
+        return {'log': {'accuracy': accuracy}}
+
+    def configure_optimizers(self):
+        return getattr(optim, self._optimizer_name)(self._model.parameters(), lr=self._lr)
+
+    @pl.data_loader
+    def train_dataloader(self):
+        return torch.utils.data.DataLoader(
+            datasets.MNIST(DIR, train=True, download=True, transform=transforms.ToTensor()),
+            batch_size=BATCHSIZE, shuffle=True)
+
+    @pl.data_loader
+    def val_dataloader(self):
+        return torch.utils.data.DataLoader(
+            datasets.MNIST(DIR, train=False, download=True, transform=transforms.ToTensor()),
+            batch_size=BATCHSIZE, shuffle=False)
+
+
+def objective(trial):
+    # PyTorch Lightning will try to restore model parameters from previous trials if checkpoint
+    # filenames match. Therefore, the filenames for each trial must be made unique.
+    checkpoint_callback = pl.callbacks.ModelCheckpoint(
+        os.path.join(DIR, 'results', 'trial_{}'.format(trial.number)))
+
+    # The default logger in PyTorch Lightining writes to event files to be consumed by
+    # TensorBoard. We create a simple logger instead that holds the log in memory so that the
+    # final accuracy can be obtained after optimization. When using the default logger, the
+    # final accuracy could be stored in an attribute of the `Trainer` instead.
+    logger = DictLogger()
+
+    trainer = pl.Trainer(
+        logger=logger,
+        train_percent_check=PERCENT_TRAIN_EXAMPLES,
+        val_percent_check=PERCENT_TEST_EXAMPLES,
+        checkpoint_callback=checkpoint_callback,
+        max_nb_epochs=EPOCHS,
+        gpus=0 if torch.cuda.is_available() else None,
+    )
+
+    # Generate the model.
+    model = Net(trial)
+
+    # Sample optimizer parameters.
+    optimizer_name = trial.suggest_categorical('optimizer', ['Adam', 'RMSprop', 'SGD'])
+    lr = trial.suggest_uniform('lr', 1e-5, 1e-1)
```

Comment:
Please use `suggest_loguniform` instead of `suggest_uniform` because the exponent part tends to be more important for learning rate tuning than the mantissa part.

BTW, @smly may suggest that we stop tuning the learning rate (see [this comment](https://github.com/pfnet/optuna/pull/559#discussion_r331818411)), but I'm not sure if it is applicable to this example.

Reply:
Changed to `suggest_loguniform`. I can also fix the optimizer in this example too if that's the preferred. It'll make the example simpler and maybe it's still clear to the reader how this example could be extended to suggest different optimizers in the trials.

Reply:
> I can also fix the optimizer in this example too if that's preferred.

I think fixing the optimizer is preferred in this example. This is partly because this example has more than 100 lines and is a bit complex compared with the other examples. And partly because we need to add more lines if we keep the learning rate tuning.

In this example, `lr` is shared by different kinds of optimizers. But, we can use separated variables such as `lr_adam` and `lr_rmsprop` as shown in the Chainer example (https://github.com/pfnet/optuna/blob/master/examples/chainer_simple.py#L53-L66).
I think it is a good practice in order to simplify the landscape of the search space because the optimal learning rate may be different for each optimizer.

Reply:
Thanks a lot for the explanation, that makes sense. I fixed the optimizer in this PR and did some related cleanups in order to make it more similar to the official PyTorch Lightning examples.